src/contractor/contractor.cpp

```diff
@@ -366,36 +367,56 @@ EdgeID Contractor::LoadEdgeExpandedGraph(
-    boost::filesystem::ifstream input_stream(edge_based_graph_filename, std::ios::binary);
-    if (!input_stream)
-        throw util::exception("Could not load edge based graph file");
+
+    const boost::interprocess::file_mapping edge_based_graph_mapping{
+        edge_based_graph_filename.c_str(), boost::interprocess::read_only};
+    boost::interprocess::mapped_region edge_based_graph_region{edge_based_graph_mapping,
+                                                               boost::interprocess::read_only};
+    edge_based_graph_region.advise(boost::interprocess::mapped_region::advice_sequential);
+    std::size_t edge_based_graph_pos = 0;
 
     const bool update_edge_weights = !segment_speed_filenames.empty();
     const bool update_turn_penalties = !turn_penalty_filenames.empty();
 
-    boost::filesystem::ifstream edge_segment_input_stream;
-    boost::filesystem::ifstream edge_fixed_penalties_input_stream;
+    const boost::interprocess::mapped_region edge_penalty_region = [&] {
+        if (update_edge_weights || update_turn_penalties)
+        {
+            const boost::interprocess::file_mapping mapping{edge_penalty_filename.c_str(),
+                                                            boost::interprocess::read_only};
+            boost::interprocess::mapped_region region{mapping, boost::interprocess::read_only};
+            region.advise(boost::interprocess::mapped_region::advice_sequential);
+            return region;
+        }
+        return boost::interprocess::mapped_region();
+    }();
 
-    if (update_edge_weights || update_turn_penalties)
-    {
-        edge_segment_input_stream.open(edge_segment_lookup_filename, std::ios::binary);
-        edge_fixed_penalties_input_stream.open(edge_penalty_filename, std::ios::binary);
-        if (!edge_segment_input_stream || !edge_fixed_penalties_input_stream)
+    const boost::interprocess::mapped_region edge_segment_region = [&] {
+        if (update_edge_weights || update_turn_penalties)
         {
-            throw util::exception("Could not load .edge_segment_lookup or .edge_penalties, did you "
-                                  "run osrm-extract with '--generate-edge-lookup'?");
+            const boost::interprocess::file_mapping mapping{edge_segment_lookup_filename.c_str(),
+                                                            boost::interprocess::read_only};
+            boost::interprocess::mapped_region region{mapping, boost::interprocess::read_only};
+            region.advise(boost::interprocess::mapped_region::advice_sequential);
+            return region;
         }
-    }
+        return boost::interprocess::mapped_region();
+    }();
 
     const util::FingerPrint fingerprint_valid = util::FingerPrint::GetValid();
-    util::FingerPrint fingerprint_loaded;
-    input_stream.read((char *)&fingerprint_loaded, sizeof(util::FingerPrint));
+    const util::FingerPrint fingerprint_loaded = *(reinterpret_cast<const util::FingerPrint *>(
```

Comment:
In the following you're using `get_address` a lot for first / last pointers.
Maybe store const first / last pointers once and then use those.